src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.SignedNumber.cs

```diff
@@ -404,6 +406,8 @@ private void WriteNumberIndented(ReadOnlySpan<byte> escapedPropertyName, long va
                 output[BytesPending++] = JsonConstants.ListSeparator;
             }
 
+            Debug.Assert(Options.SkipValidation || _tokenType != JsonTokenType.PropertyName);
```

Comment:
this assert shows up in a really a lot of places, does it make sense to factor this out? If the logic changed in the future would all of these assets change together or separately?

Reply:
We have an issue to refactor some of the writer code to avoid the duplication, so I would defer to that and do a global cleanup rather than avoiding duplication in this local instance. This is primarily why I decided to keep the duplication for now. Also, given this is `Debug.Assert` code, to me it is clearer to have it be inline and also helps with the comparison between the WriteProperty and WriteValue API implementations (where one has an branch on `(_tokenType != JsonTokenType.PropertyName)`, while this does not), and I wanted to make the distinction clearer.